docs/visualizing-your-sql-queries/plotting-with-seaborn.md

```diff
@@ -1,14 +1,561 @@
+
+```{code-cell} ipython3
+%%sql
+CREATE SCHEMA s1;
+CREATE TABLE s1.account AS
+FROM read_csv_auto('expanded_data/account.csv', header=True, sep=',');
+CREATE TABLE s1.district AS
+FROM read_csv_auto('expanded_data/district.csv', header=True, sep=',');
+CREATE TABLE s1.loan AS
+FROM read_csv_auto('expanded_data/loan.csv', header=True, sep=',');
+```
+
+The code above will create three tables in the database schema: `s1.account`, `s1.district`, `s1.loan`.
+
+## Exploring the data
+
+Let's take a look at its entries.
+
+```{code-cell} ipython3
+%sqlcmd explore --table s1.account
+```
+
+```{code-cell} ipython3
+%sqlcmd explore --table s1.district
+```
+
+```{code-cell} ipython3
+%sqlcmd explore --table s1.loan
+```
+
+## Matplotlib inheritance
+
+Seaborn is built on top of Matplotlib. Therefore, depending on the seaborn plotting command, it will return either a Matplotlib axes or figure object. If the plotting function is axes-level, a single `matplotlib.pyplot.Axes` object is returned. This object accepts an `ax=` argument, which integrates with Matplotlib's <b>object-oriented interface</b> and allows composing plots with other plots. On the other hand, if the plotting function is figure-level, a `FacetGrid` object is returned. This object, unlike the axes-level object, is more standalone, but "smart" about subplot organization. To learn about these objects in greater detail, visit seaborn's website [here](https://seaborn.pydata.org/tutorial/function_overview.html#figure-level-vs-axes-level-functions).
+
+A few examples denoting this distinction are shown below.
+
+### Axes-level
+
+Suppose we want to identify whether gentrification increases the average salary of two regions, using the data we downloaded above. We first save our [CTE (Common Table Expression)](https://ploomber-sql.readthedocs.io/en/latest/advanced-querying-techniques/ctes) named `levels_example` that takes in the columns, `average_salary`, `ratio_of_urban_inhabitants`, and `region`, and filters for two regions, 'central Bohemia' and 'east Bohemia', from the `s1.district` table.
+
+```{code-cell} ipython3
+%%sql --save levels_example
+SELECT average_salary, ratio_of_urban_inhabitants, region
+FROM s1.district
+WHERE region IN ('central Bohemia', 'east Bohemia');
+```
+
+The result of the CTE is saved as a pandas `DataFrame()`:
+
+```{code-cell} ipython3
+result = %sql SELECT * FROM levels_example
+df = result.DataFrame()
+```
+
+You can determine what is returned using Python's `type()` function:
+
+```{code-cell} ipython3
+import seaborn as sns   # noqa: E402
```

Comment:
Theres another way to avoid linting errors rather than use `# noq: E402`. 

I had the same problem with my Plotly section and @lfunderburk helped me out. Please reference our discussion here: https://github.com/ploomber/sql/pull/59

Reply:
When I do fix the import linting errors by putting them at the top like Laura's suggestion in plotly PR, I get another `flake8` issue of `F821 undefined name 'sns'` in the cells where I assign `sns.relplot()` to a variable `g`.  My plots render fine too and did before I made this change.

Reply:
Is this issue resolved or is the error still showing up?

Reply:
I still obtain the error, not resolved yet